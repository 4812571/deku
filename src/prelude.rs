/*! Crate prelude

[What is a prelude?](https://doc.rust-lang.org/std/prelude/)
*/
pub use crate::{
    error::DekuError, DekuContainerRead, DekuContainerWrite, DekuRead, DekuUpdate, DekuWrite,
};
pub use bitvec::{
    order::BitOrder, order::Lsb0, order::Msb0, slice::BitSlice, vec::BitVec, view::BitView,
};
