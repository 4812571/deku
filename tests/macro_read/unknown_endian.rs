use deku::prelude::*;

#[derive(DekuRead)]
#[deku(endian = "1")]
struct Test1 {
    a: u8,
}

#[derive(DekuRead)]
struct Test2 {
    #[deku(endian = "2")]
    a: u8,
}

#[derive(DekuRead)]
#[deku(id_type = "u8", endian = "3")]
enum Test3 {}

#[derive(DekuRead)]
#[deku(id_type = "u8")]
enum Test4 {
    #[deku(id = "1")]
    A(#[deku(endian = "4")] u8),
}

fn main() {}
