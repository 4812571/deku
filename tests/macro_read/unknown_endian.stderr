error: Unknown endian name
 --> $DIR/unknown_endian.rs:4:17
  |
4 | #[deku(endian = "1")]
  |                 ^^^

error: Unknown endian name
  --> $DIR/unknown_endian.rs:11:21
   |
11 |     #[deku(endian = "2")]
   |                     ^^^

error: Unknown endian name
  --> $DIR/unknown_endian.rs:16:33
   |
16 | #[deku(id_type = "u8", endian = "3")]
   |                                 ^^^

error: Unknown endian name
  --> $DIR/unknown_endian.rs:23:23
   |
23 |     A(#[deku(endian = "4")] u8),
   |                       ^^^
